### PR TITLE
Added console warning for `Table` when both `subRows` and `subComponent` exist

### PR DESCRIPTION
**Pull request by smmr-dn**
## Changes

Added console warning when users pass in `subRows` and `subComponent` for the `Table` at the same time, mentioned in one of the comments in this PR: https://github.com/iTwin/iTwinUI/pull/2208.

## Testing

Verified that the warning appears when there are both `subRows` and `subComponent` in the `Table`.

## Docs

Added changeset

### CODE REVIEW

File: packages/itwinui-react/src/core/Table/Table.tsx [hunk 1]
```diff
@@ -656,6 +656,12 @@ export const Table = <
     }
   }
 
+  if (data.some((item) => (item.subRows as T[]).length > 0) && subComponent) {
```
Comment (mayank99) [inline packages/itwinui-react/src/core/Table/Table.tsx:659]: Can you also add a check for `process.env.NODE_ENV === 'development'`?

Comment (smmr-dn) [reply packages/itwinui-react/src/core/Table/Table.tsx:659]: I have added the if condition to check for the dev environment.

Comment (mayank99) [reply packages/itwinui-react/src/core/Table/Table.tsx:659]: it should be made part of the same `if` condition instead of a nested one.

Comment (smmr-dn) [reply packages/itwinui-react/src/core/Table/Table.tsx]: I have made the change.

**Review by mayank99 — APPROVED** [review]
LGTM. I verified that the warning appears when using both `subRows` and `subComponent`.




You should also mention in the PR description how you tested it.

Also, this is a user-facing change, so I think it needs a `patch` changeset.

**Comment by smmr-dn** [comment]
> LGTM. I verified that the warning appears when using both `subRows` and `subComponent`.
> 
> 
> 
> You should also mention in the PR description how you tested it.
> 
> Also, this is a user-facing change, so I think it needs a `patch` changeset.

I have modified the PR description and added a changeset.

File: packages/itwinui-react/src/core/Table/Table.tsx [hunk 2]
```diff
@@ -656,6 +656,16 @@ export const Table = <
     }
   }
 
+  if (
+    process.env.NODE_ENV === 'development' &&
+    data.some((item) => (item.subRows as T[]).length > 0) &&
+    subComponent
```
[the remark this answers is not in the document]
Comment (smmr-dn) [reply packages/itwinui-react/src/core/Table/Table.tsx]: I have made the change.

**Review by mayank99 — APPROVED** [review]

**Pull request merged**